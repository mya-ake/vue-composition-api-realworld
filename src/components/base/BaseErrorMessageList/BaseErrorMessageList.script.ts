import { createComponent } from '@vue/composition-api';

export default createComponent({
  props: {
    errors: {
      required: true,
    },
  },
});
