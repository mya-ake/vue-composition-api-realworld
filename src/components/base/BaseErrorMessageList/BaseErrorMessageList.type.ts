import { ErrorItem } from '@/types';

export namespace BaseErrorMessageListType {
  export interface Props {
    messages: ErrorItem[];
  }
}
