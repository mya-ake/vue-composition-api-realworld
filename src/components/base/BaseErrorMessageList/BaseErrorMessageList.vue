<template functional>
  <ul class="error-messages">
    <li v-for="error in props.errors" :key="error.code">
      {{ error.message }}
    </li>
  </ul>
</template>

<script lang="ts" src="./BaseErrorMessageList.script.ts"></script>
