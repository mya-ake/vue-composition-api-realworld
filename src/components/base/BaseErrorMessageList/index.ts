export { default as BaseErrorMessageList } from './BaseErrorMessageList.vue';
export * from './BaseErrorMessageList.type';
