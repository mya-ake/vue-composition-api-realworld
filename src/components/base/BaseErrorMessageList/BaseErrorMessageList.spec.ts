import { shallowMount, Wrapper } from '@vue/test-utils';
import BaseErrorMessageList from './BaseErrorMessageList.vue';

describe('components/base/BaseErrorMessageList', () => {
  describe('mountable', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(BaseErrorMessageList, {
        propsData: { errors: [{ code: 'test', message: 'error message' }] },
      });
    });

    it('is vue functional component', () => {
      expect(wrapper.isVueInstance()).toBe(false);
      // @ts-ignore
      expect(wrapper.isFunctionalComponent).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
