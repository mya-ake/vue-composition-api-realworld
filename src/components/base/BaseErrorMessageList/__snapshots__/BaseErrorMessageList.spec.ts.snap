// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/base/BaseErrorMessageList mountable snapshot 1`] = `
<ul
  class="error-messages"
>
  <li>
    
    error message
  
  </li>
</ul>
`;
